### PR TITLE
bazel: add staticcheck linters

### PR DESCRIPTION
**Pull request by AlexTereshenkov**
Adopted from https://github.com/sluongng/nogo-analyzer/tree/master/examples/staticcheck

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -20,15 +29,25 @@ http_archive(
...
 gazelle_dependencies()
 
 load("//:deps.bzl", "go_dependencies")
 
 # gazelle:repository_macro deps.bzl%go_dependencies
 go_dependencies()
+
+load("@com_github_sluongng_nogo_analyzer//staticcheck:deps.bzl", "staticcheck")
+
+staticcheck()
```
Comment (ekhabarov) [inline WORKSPACE:53]: This should be called before the first usage of nogo. Move it to line 38.

**Review by ekhabarov — CHANGES_REQUESTED** [review]

File: WORKSPACE (hunk 1 above)
Comment (AlexTereshenkov) [reply WORKSPACE]: Done, thank you! Couldn't figure out why it works here https://github.com/sluongng/nogo-analyzer/blob/master/examples/staticcheck/WORKSPACE#L54 :(

File: cmd/dependencies.go [hunk 2]
```diff
@@ -24,7 +25,10 @@ func dependencies(filePath string, targets []string, transitive bool, reflexive
 		return nil, err
 	}
 	var deps []string
-
+	_, _ = time.Parse("01-01-2023", "2023-01-01")
+	if !!true {
+		time.Sleep(1)
+	}
```
Comment (AlexTereshenkov) [inline cmd/dependencies.go:31]: @ekhabarov having these lines doesn't trigger any errors from the linter:

```
$ bazel build //...
INFO: Analyzed 7 targets (1 packages loaded, 5 targets configured).
INFO: Found 7 targets...
INFO: Elapsed time: 1.227s, Critical Path: 0.28s
INFO: 9 processes: 2 internal, 7 linux-sandbox.
INFO: Build completed successfully, 9 total actions
```

I must be missing something in the configuration? :/

File: go_deps.bzl [hunk 3]
```diff
@@ -0,0 +1,4 @@
+load("@bazel_gazelle//:deps.bzl", "go_repository")
+
+def go_deps():
+  pass
```
Comment (AlexTereshenkov) [inline go_deps.bzl:4]: Copypasta from the example in the repo. What's the purpose of this, is it needed at all?

File: cmd/dependencies.go (hunk 2 above)
Comment (AlexTereshenkov) [reply cmd/dependencies.go:31]: staticcheck does report the issues:

```
$ staticcheck ./...
cmd/dependencies.go:28:20: parsing time "01-01-2023" as "01-01-2023": cannot parse "" as "3" (SA1002)
cmd/dependencies.go:29:5: negating a boolean twice has no effect; is this a typo? (SA4013)
cmd/dependencies.go:30:14: sleeping for 1 nanoseconds is probably a bug; be explicit if it isn't (SA1004)
🚨 Error: The command exited with status 1
```

File: WORKSPACE (hunk 1 above)
Comment (ekhabarov) [reply WORKSPACE]: Looks like I was wrong a bit, `staticcheck()`should be called before `gazelle_dependencies()`

```diff
diff --git a/WORKSPACE b/WORKSPACE
index 59c7655..0ee63a0 100644
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -36,16 +36,16 @@ load("@io_bazel_rules_go//go:deps.bzl", "go_register_toolchains", "go_rules_depe
 
 go_rules_dependencies()
 
-# This should be called before the first usage of nogo;
-# @co_honnef_go_tool is defined by `staticcheck()` macro
-load("@com_github_sluongng_nogo_analyzer//staticcheck:deps.bzl", "staticcheck")
-staticcheck()
-
 go_register_toolchains(
     nogo = "@//:nogo",
     version = "1.22.5",
 )
 
+# This should be called before the first usage of nogo;
+# @co_honnef_go_tool is defined by `staticcheck()` macro
+load("@com_github_sluongng_nogo_analyzer//staticcheck:deps.bzl", "staticcheck")
+staticcheck()
+
 gazelle_dependencies()
 
 load("//:deps.bzl", "go_dependencies")

```

File: go_deps.bzl (hunk 3 above)
Comment (ekhabarov) [reply go_deps.bzl:4]: Go dependencies are usually listed here, but you have `deps.bzl` file for this. Where to write dependencies defined [here](https://github.com/AlexTereshenkov/dg-query/blob/bazel/add-staticcheck-linters/BUILD.bazel#L31).

Comment (AlexTereshenkov) [reply go_deps.bzl:4]: I see, thanks. Removed these bits then as they are redundant.

File: cmd/dependencies.go (hunk 2 above)
Comment (AlexTereshenkov) [reply cmd/dependencies.go]: Fixed by moving the `staticcheck()` before `gazelle_dependencies()`

**Pull request merged**